skill-tree /src/main/java/com/RDS/skilltree/Exceptions/NoEntityException.java

```diff
@@ -0,0 +1,12 @@
+package com.RDS.skilltree.Exceptions;
+
+public class NoEntityException extends RuntimeException {
```

Comment:
Is there a reason why this is a `RuntimeException` and not `Exception`?

Reply:
An Exception is checked, and a RuntimeException is unchecked.

Checked means that the compiler requires that you handle the exception in a catch, or declare your method as throwing it (or one of its superclasses).

Generally, throw a checked exception if the caller of the API is expected to handle the exception, and an unchecked exception if it is something the caller would not normally be able to handle, such as an error with one of the parameters, i.e. a programming mistake.

So here we are using these error to throw error for things the caller API couldn't have checked before hand like, "no user foiund for this user id " etc

Reply:
@bhtibrewal  and @debanjanc01 what about overwriting the stack trace in case of check and un-checked (question for both)

Reply:
@iamitprakash if you're talking about  custom exceptions, then in my opinion it depends on how the exception is being used. 

If it's Exceptions that ultimately propagates directly to the Controller and is handled by a global exception handler, then the stack trace can be handled. The assumption here is that the exception handler will be sending the correct response to the API caller. 

If it's Exceptions that are being used internally, and handled via Catch statements, we might want to error log with the stacktrace.